lib/common/client/concerns/session_based_client.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+module Common
+  module SessionBasedClient
+    extend ActiveSupport::Concern
+
+    attr_reader :config, :session
+
+    def authenticate
+      binding.pry
+      if session.expired?
+        session = get_session
+        session.save
+      end
+      self
+    end
+
+    private
+
+    def auth_headers
+      config.base_request_headers.merge('appToken' => config.app_token, 'mhvCorrelationId' => session.user_id.to_s)
```

Comment:
@austin-martinez that's why there is no client.rb class. You still create the class, but you mixin whatever you need. As for Configuration, yes those are the defaults, we can move it to a mixin as well or alternately you can just override the method base_request_header.